etl/src/energuide/embedded/heating.py

```diff
@@ -0,0 +1,151 @@
+                                                   french='Plinthe électrique'),
+        HeatingType.FURNACE: bilingual.Bilingual(english='Furnace', french='Fournaise'),
+        HeatingType.BOILER: bilingual.Bilingual(english='Boiler', french='Chaudière'),
+        HeatingType.COMBO_HEAT_DHW: bilingual.Bilingual(english='Combo Heat/DHW',
+                                                        french='Combinaison chaleur/ Eau Chaude Domestique'),
+        HeatingType.P911: bilingual.Bilingual(english='Certified combo system, space and domestic water heating',
+                                              french='Système combiné certifié pour le chauffage '
+                                                     'des locaux et de l’eau'),
+    }
+
+    _ENERGY_SOURCE_CODES = {
+        2: EnergySource.NATURAL_GAS,
+        3: EnergySource.OIL,
+        4: EnergySource.PROPANE,
+        5: EnergySource.WOOD,
+    }
+
+    _ENERGY_SOURCE_TRANSLATIONS = {
+        EnergySource.ELECTRIC: bilingual.Bilingual(english='Electric Space Heating',
+                                                   french='Chauffage électrique'),
+        EnergySource.NATURAL_GAS: bilingual.Bilingual(english='Natural Gas', french='Chauffage au gaz naturel'),
+        EnergySource.OIL: bilingual.Bilingual(english='Oil Space Heating', french='Chauffage au mazout'),
+        EnergySource.PROPANE: bilingual.Bilingual(english='Propane Space Heating', french='Chauffage au propane'),
+        EnergySource.WOOD: bilingual.Bilingual(english='Wood Space Heating (Mixed Wood, Hardwood, '
+                                                       'Soft Wood or Wood Pellets)',
+                                               french='Chauffage au bois(Bois mélangé, Bois dur, Bois mou, '
+                                                      'Granules de bois)'),
+    }
+
+    @classmethod
+    def _get_output_size(cls, node: element.Element) -> float:
+        capacity_node = node.find('Type1/*/Specifications/OutputCapacity')
+        assert capacity_node is not None
+
+        units = capacity_node.attrib['uiUnits']
+        capacity_value = float(capacity_node.attrib['value'])
+        capacity: typing.Optional[float]
+        if units == 'kW':
+            capacity = capacity_value
+        elif units == 'btu/h':
+            capacity = capacity_value / cls._KWH_TO_BTU
+        else:
+            raise InvalidEmbeddedDataTypeError(
+                Heating, f'Unknown capacity units: f{units}')
+        assert capacity is not None
+        return capacity
+
+    @classmethod
+    def _get_heating_type(cls, node: element.Element) -> HeatingType:
+        candidates = [candidate.tag for candidate in node.xpath('Type1/*')]
+        heating_type: typing.Optional[HeatingType] = None
+        for candidate in candidates:
+            if candidate in cls._HEATING_TYPE_NODE_NAMES:
+                heating_type = cls._HEATING_TYPE_NODE_NAMES[candidate]
+                break
```

Comment:
This section seems odd to me, I'm guessing there's a good reason for it, possibly motivated by lack of test data?

Reply:
Yes, we don't know what all the heating type node names are. There are multiple possible child nodes of `Type1`, only one of them is the one we want, and we don't know all the names. The three we know are `Baseboards`, `Furnace`, and `Boiler`. I haven't been able to figure out the other 2 yet, so I intentionally want it to fail and raise an error.